tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2028,6 +2028,79 @@ def _check_import(self, name, node):
                 msgid="no-raise-with-traceback", node=node, confidence=None
             )
 
+class NameExceedsStandardCharacterLength(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that the character length of type and property names are not over 40 characters."""
+    name = "name-too-long"
+    priority = -1
+    msgs = {
+        "C4751": (
+            "Name is over standard character length of 40.",
+            "name-too-long",
+            "Only use names that are less than 40 characters."
+        ),
+    }
+
+    STANDARD_CHARACTER_LENGTH = 40
+
+    def visit_classdef(self,node):
+        """Visit every class and check that the
+         class name is within the character length limit."""
+
+        try:
+            self.iterate_through_names(node, True)    
+        except:
+            pass   
+
+
+    def visit_functiondef(self, node):
+        """Visit every function and check that the function and 
+        its variable names are within the character length limit."""
+        try:
+            self.iterate_through_names(node, False)    
+        except:
+            pass      
+
+    def iterate_through_names(self, node, ignore_function):
```

Comment:
helper function called by both visit_function and visit_class, so we check that if it is a classdef call to the helper function that we don't double check functions.

Reply:
what is ignore_function? I can't find where it's defined

Reply:
Added it~ `:param ignore_function: Whether the function is being called 
             from `visit_classdef` or not. If it is called from `visit_classdef`
             ignore_function should be `True` to avoid repeat warnings on functions. `